src/VirtualClient/VirtualClient.Actions/MLPerfTraining/MLPerfTrainingExecutor.cs

```diff
@@ -0,0 +1,515 @@
+        /// The output directory of MLPerf.
+        /// </summary>
+        protected string OutputDirectoryPath
+        {
+            get
+            {
+                return this.PlatformSpecifics.Combine(this.ExecutionDirectoryPath, "results");
+            }
+        }
+
+        /// <summary>
+        /// Initializes the environment for execution of the MLPerf Training workload.
+        /// </summary>
+        protected override async Task InitializeAsync(EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            this.Logger.LogTraceMessage($"{this.TypeName}.InitializationStarted", telemetryContext);
+
+            this.ThrowIfPlatformNotSupported();
+
+            await this.ThrowIfUnixDistroNotSupportedAsync(cancellationToken)
+                .ConfigureAwait(false);
+
+            MLPerfState state = await this.stateManager.GetStateAsync<MLPerfState>($"{nameof(MLPerfState)}", cancellationToken)
+                ?? new MLPerfState();
+
+            if (!state.Initialized)
+            {
+                // add user in docker group
+                await this.ExecuteCommandAsync("usermod", $"-aG docker {this.Username}", this.ExecutionDirectoryPath, cancellationToken);
+
+                // Setup Environment
+                await this.SetupEnvironmentAsync(cancellationToken);
+                state.Initialized = true;
+                await this.stateManager.SaveStateAsync<MLPerfState>($"{nameof(MLPerfState)}", state, cancellationToken);
+            }
+        }
+
+        /// <summary>
+        /// Creates setup for MLPerf Training workload.
+        /// </summary>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        protected async Task SetupEnvironmentAsync(CancellationToken cancellationToken)
+        {
+            string dockerImageCommand = $"docker build --pull -t {this.GetContainerName()} .";
+            string dockerRunCommand = $"docker run --runtime=nvidia {this.GetContainerName()}";
+
+            await this.ExecuteCommandAsync(
+                "sudo",
+                dockerImageCommand,
+                this.ExecutionDirectoryPath,
+                cancellationToken);
+
+            await this.ExecuteCommandAsync(
+                "sudo",
+                dockerRunCommand,
+                this.ExecutionDirectoryPath,
+                cancellationToken);
+        }
+
+        /// <summary>
+        /// Gets the container name created by MLPerf Training.
+        /// </summary>
+        /// <returns>Container name created by MLPerf Training</returns>
+        /// <exception cref="WorkloadException"></exception>
+        protected string GetContainerName()
+        {
+            // Update this function to accomodate other architectures
+            if (this.Platform == PlatformID.Unix && this.CpuArchitecture == Architecture.X64)
+            {
+                return $"mlperf-training-{this.Username}-x86_64:{this.ContainerName}";
```

Comment:
who maintains these containers? I know mlperf has a bunch of containers, just making sure we have a way to update it if needs be

Reply:
The docker containers are hosted locally.
Each benchmark in the workload uses a separate container to run the workload, due to the dependencies they require while building the container.